src/main/java/com/github/minecraftschurlimods/arsmagicalegacy/common/item/spellbook/SpellBookContainer.java

```diff
@@ -0,0 +1,115 @@
+package com.github.minecraftschurlimods.arsmagicalegacy.common.item.spellbook;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+import net.minecraft.nbt.Tag;
+import net.minecraft.world.Container;
+import net.minecraft.world.SimpleContainer;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+
+import java.util.List;
+
+public record SpellBookContainer(ItemStack stack, SimpleContainer active, SimpleContainer back) implements Container {
```

Comment:
This file needs general cleanup.

- Remove the empty line after this line.
- Remove the empty lines inside the methods (they don't improve readability at all)
- Make the `if () { return }` statements single-line ifs, or even better, use ?:
- Press Ctrl+Alt+L please.